### PR TITLE
play: upgrade --start-task's specified in legacy format

### LINKED ISSUE #4920: Start task does not accept legacy `<task>.<point>` format
```
$ cylc play simple-c7-datetime/ --start-task=foo.20020830 -n

WARNING - Backward compatibility mode ON ...
ERROR - Workflow shutting down - PointParsingError: Incompatible value for <class 'cylc.flow.cycling.iso8601.ISO8601Point'>:
    foo.20020830: Invalid ISO 8601 date representation: foo.20020830
INFO - DONE
```

This may be particularly annoying because the workflow has now started up so you'd have to clean the DB before trying again.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Extend CLI upgrading to the Cylc 8 --start-task option.
* Closes #4920

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Does not need tests (why?).
- [x] No change log entry required (why? minor edge case of little concern)
- [x] No documentation update required.

### CODE REVIEW

File: tests/functional/cylc-play/10-start-task-upgrade.t [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+set_test_number 2
+
+run_fail "${TEST_NAME_BASE}" \
+    cylc play Agrajag --start-task foo.123 --start-task bar.234
+
+grep_ok \
+    'Cylc7 format is deprecated, using: 123/foo 234/bar' \
+    "${TEST_NAME_BASE}.stderr"
+
```
Comment (MetRonnie) [inline tests/functional/cylc-play/10-start-task-upgrade.t:31]:
```suggestion

purge
```

Comment (hjoliver) [reply tests/functional/cylc-play/10-start-task-upgrade.t:31]: There's no workflow installed, so nothing to purge.

**Review by hjoliver — APPROVED** [review]
Looks good, tested as working :+1:

**Review by MetRonnie — APPROVED** [review]
